backend/Social/SocialAPI.py

```diff
@@ -420,24 +443,24 @@ def posts():
 
                 response = []
 
-                userIDList = [x["userID"] for x in data]
-                profiles = list(db.Profiles.find({"userID": {"$in": userIDList}}))
+                userIDList = [post["userID"] for post in data]
+                profiles = list(db.Profiles.find({"userID": {"$in": userIDList}}, {"_id": 0}))
                 profileDict = {}
-                for x in profiles:
-                    profileDict[x["userID"]] = x
+                for profile in profiles:
+                    profileDict[profile["userID"]] = profile
 
                 # for item in data:
                 #     response.append(item)
                 for item in data:
                     profile = profileDict.get(item["userID"])
+                    imageKey = profile.get('imageKey') if profile != None else None
```

Comment:
(When a try-except block is added around the ```S3.app.read()``` code, this part of code outputs legitimate responses according to my test.)